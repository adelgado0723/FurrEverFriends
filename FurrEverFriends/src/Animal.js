import React from 'react';
import { Link } from '@reach/router';

class Animal extends React.Component {
  render() {
    const { name, species, breed, media, location, id } = this.props;
    let thumbnail = {};
    if (media && media[0] && media[0].urlSecureThumbnail) {
      thumbnail = media[0].large.url;
    }

    return (
      <Link to={`/details/${id}`} state={this.props} className="animal">
        <div className="image-container">
          <img src={thumbnail} alt={name} />
        </div>
        <h1>{name}</h1>
        <h2>{`${species} - ${breed} - ${
          location && location.formattedAddress ? location.formattedAddress : ''
        }`}</h2>
      </Link>
    );
  }
}

export default Animal;
