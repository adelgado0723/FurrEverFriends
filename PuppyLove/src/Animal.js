import React from "react";

class Animal extends React.Component {
  render() {
    const { name, species, breed, media, location } = this.props;
    let thumbnail = {};
    if (media && media[0] && media[0].urlSecureThumbnail) {
      thumbnail = media[0].urlSecureThumbnail;
    }

    return (
      <div className="animal" >
        <div className="image-container">
          <img src={thumbnail} alt={name} />
        </div>
        <div className="info">
          <h1>{name}</h1>
          <h2>{`${species} - ${breed} - ${location}`}</h2>
        </div>
      </div>
    );

  }
}

export default Animal;