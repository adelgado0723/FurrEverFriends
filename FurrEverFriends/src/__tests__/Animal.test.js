import React from 'react';
import { create } from 'react-test-renderer';
import Animal from '../Animal';

test('snapshot', () => {
  const c = create(<Animal />);
  expect(c.toJSON()).toMatchSnapshot();
});
