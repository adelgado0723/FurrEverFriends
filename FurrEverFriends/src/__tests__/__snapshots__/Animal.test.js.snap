// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<a
  className="animal"
  href="/details/undefined"
  onClick={[Function]}
>
  <div
    className="image-container"
  >
    <img
      src={Object {}}
    />
  </div>
  <h1 />
  <h2>
    undefined - undefined - 
  </h2>
</a>
`;
